src/Classes/MyRadio/CoreUtils.php

```diff
@@ -257,10 +257,17 @@ public static function encodeTrack($tmpfile, $dbfile)
         } elseif (!file_exists($dbfile.'.mp3') || !file_exists($dbfile.'.ogg')) {
             throw new MyRadioException('Conversion failed', 500);
         }
-
-        rename($tmpfile, "{$dbfile}.mp3.orig");
-        if (!file_exists("{$dbfile}.mp3.orig")) {
-            throw new MyRadioException('Could not move file to library.');
+        $orig_new_filename = $dbfile .".mp3.orig";
+        copy($tmpfile, $orig_new_filename);
```

Comment:
Maybe a comment or two about why it's copy & unlink?

Reply:
Oh, I put it in the relevant commit comment, i'll copy it over.